### PR TITLE
Merge release v0.9.0 into main

### PR DESCRIPTION
**Pull request by joeycarter**
Final merge of the v0.9.0 release branch into main.

### CODE REVIEW

File: .dep-versions [hunk 1]
```diff
@@ -8,11 +8,11 @@ enzyme=v0.0.149
 
 # For a custom PL version, update the package version here and at
 # 'doc/requirements.txt
-pennylane=0.39.0.dev30
+pennylane=0.39.0
```
Comment (rmoyard) [inline .dep-versions:11]: I think it should not be pinned

File: .dep-versions [hunk 2]
```diff
@@ -8,11 +8,11 @@ enzyme=v0.0.149
...
 
 # For a custom LQ/LK version, update the package version here and at
 # 'doc/requirements.txt'. Also, update the 'LIGHTNING_GIT_TAG' at
 # 'runtime/Makefile' and at all GitHub workflows, using the exact
 # commit hash corresponding to the merged PR that implements the
 # desired feature.
-lightning=0.39.0-dev38
+lightning=0.39.0
```
Comment (rmoyard) [inline .dep-versions:18]: Same

File: .dep-versions (hunk 1 above)
Comment (joeycarter) [reply .dep-versions:11]: Chatted offline, it shouldn't make a difference for the time being so we'll leave this as is.

File: .dep-versions (hunk 2 above)
Comment (joeycarter) [reply .dep-versions:18]: See comment above.

**Comment by joeycarter** [comment]
There's an issue with readthedocs that's affecting everyone. Stand by...

**Review by rmoyard — APPROVED** [review]
:1st_place_medal:

**Pull request merged**